streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -605,13 +613,14 @@ private long sumOfChangelogOffsets(final TaskId id, final Map<TopicPartition, Lo
             final long offset = changelogEntry.getValue();
 
 
-            if (offset == Task.LATEST_OFFSET) { // this condition can only be true for active tasks; never for standby
+            if (offset == Task.LATEST_OFFSET) {
+                // this condition can only be true for active tasks; never for standby
                 // for this case, the offset of all partitions is set to `LATEST_OFFSET`
                 // and we "forward" the sentinel value directly
                 return Task.LATEST_OFFSET;
             } else {
                 if (offset < 0) {
-                    throw new IllegalStateException("Offset should not be negative.");
+                    throw new IllegalStateException("Expected not to get a sentinel offset, but got: " + changelogEntry);
```

Comment:
I independently implemented this check before Matthias's PR, and preferred this message.